backend/fms_core/models/dataset.py

```diff
@@ -0,0 +1,47 @@
+from django.db import models
+from django.core.exceptions import ValidationError
+import reversion
+
+from .tracked_model import TrackedModel
+from .project import Project
+from .experiment_run import ExperimentRun
+from .container import Container
+
+from ._constants import STANDARD_NAME_FIELD_LENGTH
+
+@reversion.register()
+class Dataset(TrackedModel):
+    """ Class to store information about the datasets of data deliveries. """
+
+    project_name = models.CharField(max_length=STANDARD_NAME_FIELD_LENGTH, help_text="Project name.")
+
+    run_name = models.CharField(max_length=STANDARD_NAME_FIELD_LENGTH, help_text="Run name.")
+
+    lane = models.PositiveIntegerField(help_text="Coordinates of the lane in a container")
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=["project_name", "run_name", "lane"], name="unique_dataset")
```

Comment:
use a more descriptive name for the constraint so we know what it represent (it is typical to list the unique fields e.g. "unique_externalprojectid_runname_lane")